src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,465 @@
+		private String version = null;
+		private ResourceType resourceType = ResourceType.DATASET;
+		private List<String> comments = null;
+		private final List<Contributor> contributors;
+		private List<EventDate> dates = null;
+		private List<FundingReference> funding = null;
+		private List<PermanentID> relatedIdentifiers = null;
+		private final List<Title> titles;
+		private List<String> errorList = new ArrayList<>();
+
+		private Builder(
+				final String identifier,
+				final List<Contributor> contributors,
+				final List<Title> titles) {
+
+			String checkedPid = identifier;
+			try {
+				checkedPid = Common.checkPid(identifier, "identifier", false);
+			} catch (IllegalArgumentException e) {
+				this.errorList.add(e.getMessage());
+			}
+			this.identifier = checkedPid;
+
+			// must be at least one contributor
+			this.contributors = Common.dedupeSimpleList(contributors);
+			if (this.contributors == null || this.contributors.isEmpty()) {
+				this.errorList.add("at least one contributor must be provided");
+			}
+
+			// must be at least one title
+			this.titles = Common.dedupeSimpleList(titles);
+			if (this.titles == null || this.titles.isEmpty()) {
+				this.errorList.add("at least one title must be provided");
+			}
+		}
+
+		/**
+		 * Sets the resource type for the resource.
+		 *
+		 * As there is currently only one resource type, entering null will set the
+		 * resource type value to DEFAULT_RESOURCE_TYPE in the resultant object.
+		 *
+		 * @param resourceType
+		 *                resource type as a string
+		 * @return this builder
+		 */
+		public Builder withResourceTypeString(final String resourceType) {
+			if (resourceType != null) {
+				try {
+					this.resourceType = ResourceType.getResourceType(resourceType);
+				} catch (IllegalArgumentException e) {
+					// TEMPORARY HACK
+					// As there is only one resource type at present
+					// (ResourceType.DATASET), null input in the resourceType
+					// field will be automatically set to that value.
+					// The error message "resourceType cannot be null or
+					// whitespace only" thus needs to be edited to say
+					// "resourceType cannot be whitespace only".
+					String errorMessage = e.getMessage();
+					if ("resourceType cannot be null or whitespace only"
+							.equals(errorMessage)) {
+						errorMessage = "resourceType cannot be whitespace only";
+					}
+					errorList.add(errorMessage);
+				}
+			}
+			else {
+				this.resourceType = DEFAULT_RESOURCE_TYPE;
+			}
+			return this;
+		}
+
+		/**
+		 * Sets the resource type for the resource.
+		 *
+		 * As there is currently only one resource type, entering null will set the
+		 * resource type value to DEFAULT_RESOURCE_TYPE in the resultant object.
+		 *
+		 * @param resourceType
+		 *                resource type as a {@link ResourceType}
+		 * @return this builder
+		 */
+		public Builder withResourceType(final ResourceType resourceType) {
+			// TEMPORARY HACK - only one resource type at present
```

Comment:
Again, I don't think this is a temporary hack, it's perfectly fine long term. The only thing I'd consider is ignoring null rather than resetting to default, but I'm not religious there.

Reply:
N/A

Reply:
:+1: